src/protocol/ipa/mod.rs

```diff
@@ -129,6 +146,140 @@ impl<F: Field + Sized> Resharable<F> for IPAModulusConvertedInputRow<F> {
     }
 }
 
+#[async_trait]
+impl<F: Field + Sized> Resharable<F> for IPAMaliciousModulusConvertedInputRow<F> {
+    type Share = MaliciousReplicated<F>;
```

Comment:
Is there any other way to avoid this duplication of code for just declaring a Share type as Malicious? @akoshelev @martinthomson ?

Reply:
The way we have done this in the past is to have `IpaModulusConvertedInputRow` be generic over `S: SecretSharing` (and probably also `F: Field`).  Then you can implement `Resharable<F> for IpaModulusConvertedInputRow<S, F>`.

This line becomes `type Share = S`, I think.  The interaction between type constraints and associated types is still something I'm not 100% on.